pkg/cache/cluster.go

```diff
@@ -702,6 +741,51 @@ func (c *clusterCache) processApi(client dynamic.Interface, api kube.APIResource
 	return nil
 }
 
+// checkPermission runs a self subject access review to check if the controller has permissions to list the resource
+func (c *clusterCache) checkPermission(ctx context.Context, reviewInterface authType1.SelfSubjectAccessReviewInterface, api kube.APIResourceInfo) (keep bool, err error) {
+	sar := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Namespace: "*",
+				Verb:      "list", // uses list verb to check for permissions
```

Comment:
Is `list` sufficient? Or would we also need to check for `get` and `watch`?

Reply:
Also, the error response after which this access review is made is a list call thus it makes sense to use list verb to confirm the access error.

Reply:
Makes sense.